### PR TITLE
[jerryscript] add port

### LINKED ISSUE #19432: [New Port Request] jerryscript
Library name:
jerryscript

Library description:
JerryScript is a lightweight JavaScript engine for resource-constrained devices such as microcontrollers.

Source repository URL:
https://github.com/jerryscript-project/jerryscript
Project homepage (if different from the source repository):

Anything else that is useful to know when adding (such as optional features the library may have that should be included):
The JERRY_MATH option must be disabled. This option includes its own implementation of sin / cos / abs and others. This is useful for microcontrollers, but not meaningful for a PC platform.

Reasons why this library should be added:
Currently vcpkg supports the following javascript engines: chakracore, duktape, v8.

chakracore - Full support for Esmascript specifications. acceptable size of binaries. But internally it uses wide-strings.

duktape - Esmacript 5 support (6 not supported). Small size of binary files.

v8 - full support for Esmascript specifications, but large binaries.

Jerryscript, in contrast to these libraries, is small in size and supports most of the features of Esmascript 6.

### PR DESCRIPTION
**Pull request by autoantwort**
Fixes #19432

### CODE REVIEW

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/jerryscript/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,30 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO jerryscript-project/jerryscript
+    REF 3bcd48f72d4af01d1304b754ef19fe1a02c96049
```
[the remark this answers is not in the document]
Comment (autoantwort) [reply ports/jerryscript/portfile.cmake:4]: It was used to test if on master https://github.com/jerryscript-project/jerryscript/issues/4750 is fixed

**Comment by LilyWangLL** [comment]
Pinging @autoantwort  for response. Is work still being done for this PR?

**Comment by LilyWangLL** [comment]
Ping @autoantwort, any updates of this PR?

**Comment by autoantwort** [comment]
I could test it again, but I currently don't have interest in this port.

**Pull request closed without merging**